### ISSUE TITLE
Node and Python extra -- more robust install

### ISSUE THREAD

**Issue by tlienart** [tlienart/Franklin.jl#558]
This is the kind of super annoying stuff to work out but it seems that some people struggle to get Franklin to work with highlight.js and/or even calling python for the minification script.

While this is irrelevant if people use Github Action or a similar containerised deployment; the people trying to deploy manually can struggle to get `optimise` to work properly. 

@mbaz kindly provided some notes on this https://github.com/tlienart/PkgPage.jl/issues/21

**Comment by aviks** [comment]
One random thought ... can minification be done using node? Then I'm struggling with one additional language, rather than two?

**Comment by tlienart** [comment]
yes so there's [`html-minifier`](https://github.com/kangax/html-minifier) but it's not completely fool proof, basically the script that is used here is very simple and doesn't really do "clever" stuff which also means that, in general, it doesn't break your stuff either. `html-minifier` by contrast does a lot of clever stuff and (last I tried) breaks a lot of stuff if you don't set it up right.

Ideally the steps would be to remove the python script and... 

1. use `purgecss` (CSS is kind of what matters most in any case after image optimization of course)
1. use `html-minifier` with the most basic setup to avoid breakage 

related #516 #388 

**Notes**: 
- if you don't manage to install the libraries fine locally, it doesn't actually matter, you'll still be able to do everything apart from optimisation but that bit is probably not very important for simple sites
- even then, if you use a deploy script via GitHub action or on GitLab, then the optimisation will work at deploy time

the one risk with this is that you can't directly test whether the optimisation phase breaks your site (though it won't in 95% of the cases).

**Comment by aviks** [comment]
>if you don't manage to install the libraries fine locally, it doesn't actually matter

Yes, the franlink runtime messages make that clear, for which I'm very grateful. I'm on windows, and even the thought of installing python or node packages gives me nightmares. My comment above was therefore mostly rhetorical!

**Comment by tlienart** [comment]
no but it's definitely something that's come up sufficiently many times that I'd like to figure it out cleanly, but it's one of those things where I can see that it's going to be hard to fix quickly and the benefit will be mostly cosmetic 🙈

**Comment by newptcai** [comment]
I cannot deploy to GitLab anymore. Maybe this is related?

The error I got is
~~~
$ pip3 install css-html-js-minify
error: externally-managed-environment
× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.11/README.venv for more information.
note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
Cleaning up project directory and file based variables 00:00
ERROR: Job failed: exit code 1
~~~

**Comment by tlienart** [comment]
@newptcai in your deploy script look for something like `optimize()` and replace it with `optimize(prerender=false, minify=false)` which will ignore the python and javascript stuff.

These dependencies will be removed in the next version of Franklin.

**Comment by fatteneder** [comment]
> I cannot deploy to GitLab anymore. Maybe this is related?
> 
> The error I got is
> 
> ```
> $ pip3 install css-html-js-minify
> error: externally-managed-environment
> × This environment is externally managed
> ╰─> To install Python packages system-wide, try apt install
>     python3-xyz, where xyz is the package you are trying to
>     install.
>     
>     If you wish to install a non-Debian-packaged Python package,
>     create a virtual environment using python3 -m venv path/to/venv.
>     Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
>     sure you have python3-full installed.
>     
>     If you wish to install a non-Debian packaged Python application,
>     it may be easiest to use pipx install xyz, which will manage a
>     virtual environment for you. Make sure you have pipx installed.
>     
>     See /usr/share/doc/python3.11/README.venv for more information.
> note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
> hint: See PEP 668 for the detailed specification.
> Cleaning up project directory and file based variables 00:00
> ERROR: Job failed: exit code 1
> ```

I managed to get it work by replacing the `before_script:` block with
```sh
before_script:
  - apt-get -qq update; apt-get -y install git python3-dev python3-pip python3-venv
  - python3 -m venv ~/.local
  - ~/.local/bin/pip3 install css-html-js-minify
```
